spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/ModuleLoader.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils.serializer
+
+import java.io._
+
+import scala.collection.JavaConverters._
+import com.google.protobuf.CodedInputStream
+import com.intel.analytics.bigdl.nn.Container
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.io.IOUtils
+import serialization.Bigdl.BigDLModule
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+
+object ModuleLoader {
+  private val hdfsPrefix: String = "hdfs:"
+
+  def loadFromFile[T: ClassTag](modelPath : String)
+    (implicit ev: TensorNumeric[T]) : AbstractModule[Activity, Activity, T] = {
+    val modelBuilder = BigDLModule.newBuilder
+    var cis : CodedInputStream = null
+    if (modelPath.startsWith(hdfsPrefix)) {
```

Comment:
I think Xin has a PR to provide a common reader utility that supports hdfs/s3/local. Please use that.